Feature filter service [#158037410]

YouKnow232 opened the pull request.

YouKnow232 committed:
Implemented filter.service [#158037410]

YouKnow232 committed:
Added two unit tests [#158037410]

JGrippo commented on projects/revaturecloud/selection/src/lib/services/filter.service.ts:
```diff
@@ -0,0 +1,23 @@
+})
+export class FilterService {
+
+  filter: Subject<SearchParameters>;
```
Lets use a behavior subject here.

JGrippo reviewed (changes requested):
Just some minor changes.

YouKnow232 committed:
Updated filter service [#158037410]

Switched Subject implementation to BehaviorSubject.
Changed internal subject var name to subject.
Changed initial behavior to use a default filter.
Changed unit testcase to account for new default behavior.

JGrippo reviewed (changes requested):
Fix the tests.

YouKnow232 committed:
Merge branch '158037410-selection-interface' into feature-filter-service

JGrippo reviewed (approved):
Nice work.
